### ISSUE TITLE
Implicit `Buffer` usage fails in Cloudflare Workers

### ISSUE THREAD

**Issue by danielgavrilov** [vercel/ai#2741]
### Description

Cloudflare Workers fail on startup if `streamText` is imported:

```ts
import { streamText } from "ai";
```

This is the error:

```
Uncaught ReferenceError: Buffer is not defined
    at null.<anonymous> (index.js:9799:16)
```

Which is pointing to this line: https://github.com/vercel/ai/blob/ddf0184b69341468898d59c5c9a0276d3e90dc76/packages/ai/core/prompt/data-content.ts#L20

The Node.js documentation recommends importing `Buffer` explicitly:

> While the `Buffer` class is available within the global scope, it is still recommended to explicitly reference it via an import or require statement.

https://nodejs.org/docs/latest-v22.x/api/buffer.html#buffer

If `Buffer` explicitly imported and Cloudflare Workers have `nodejs_compat` mode enabled, this works without issues.

**Comment by lgrammel** [comment]
Are there solutions that do not require an explicit import?

**Comment by lgrammel** [comment]
Fixed in `ai@3.3.13`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This reports a concrete runtime compatibility failure in existing AI SDK behavior, and the maintainer confirmed it was fixed.

### Evidence
- Importing `streamText` caused Cloudflare Workers to fail at startup with `ReferenceError: Buffer is not defined`.
- The failure was traced to implicit global `Buffer` usage in `packages/ai/core/prompt/data-content.ts`.
- A maintainer confirmed the issue was fixed in `ai@3.3.13`.
